crates/tiny/src/conn.rs

```diff
@@ -478,91 +478,117 @@ fn handle_irc_msg(ui: &UI, client: &dyn Client, msg: wire::Msg) {
+                }
+                // RPL_ENDOFNAMES: End of NAMES list
+                (366, _) => {}
+                 // RPL_UNAWAY or RPL_NOWAWAY
+                (305 | 306, _) => {
+                    let msg = &params[1];
+                    ui.add_client_msg(msg, &MsgTarget::AllServTabs { serv });
+                }
+                // ERR_NOSUCHNICK
+                (401, n_params) if n_params > 2 => {
+                    let nick = &params[1];
+                    let msg = &params[2];
+                    ui.add_client_msg(msg, &MsgTarget::User { serv, nick });
+                }
+                // RPL_AWAY
+                (301, n_params) if n_params > 2 => {
+                    let nick = &params[1];
+                    let msg = &params[2];
+                    ui.add_client_msg(
+                        &format!("{} is away: {}", nick, msg),
+                        &MsgTarget::User { serv, nick },
+                    );
+                }
+                // ERR_BADCHANNAME
+                (479, n_params) if n_params > 2  => {
+                    let chan = &params[1];
+                    let msg = &params[2];
+                    ui.add_client_err_msg(
+                        &format!("Unable to join {}: {}", chan, msg),
+                        &MsgTarget::Chan { chan: ChanNameRef::new(chan), serv },
+                    );
+                }
+                // ERR_CHANNELISFULL    471
+                // ERR_INVITEONLYCHAN   473
+                // ERR_BANNEDFROMCHAN   474
+                // ERR_BADCHANNELKEY    475
+                // ERR_NEEDREGGEDNICK   477
+                // ERR_THROTTLE         480
+                (471 | 473 | 474 | 475 | 477 | 480, n_params) if n_params > 2 => {
+                    let chan = &params[1];
+                    let msg = &params[2];
+                    ui.add_client_err_msg(
+                        msg,
+                        &MsgTarget::Chan { chan: ChanNameRef::new(chan), serv },
+                    );
+                }
+                _ => {
+                    match pfx {
+                        Some(Server(msg_serv)) | Some(Ambiguous(msg_serv)) => {
+                            let msg_target = MsgTarget::Server { serv };
+                            ui.add_privmsg(
+                                &msg_serv,
+                                &params.join(" "),
+                                time::now(),
+                                &msg_target,
+                                false,
+                                false,
+                            );
+                            ui.set_tab_style(TabStyle::NewMsg, &msg_target);
+                        }
+                        Some(User { .. }) | None => {
+                            debug!(
+                                "Ignoring numeric reply {}: pfx={:?}, params={:?}",
+                                n, pfx, params
+                            );
+                        }
```

Comment:
I feel like changes in this file are unrelated to this PR, right? Could you revert these please? I'm happy to review these changes in its own PR.